README.md

```diff
@@ -1 +1,467 @@
+
+## ranges
+
+Sucre syntaxique : `1..4` génère `1, 2, 3, 4`
+
+## in
+
+C'est franchement trop pratique, vérifier qu'une valeur est dans un tableau, ça devrait faire partie du langage sans avoir à faire d'appel de fonction.
+
+```
+2 in 1..6 // true
+
+c = `r`
+c in `string` // true
+```
+
+## string
+
+Si on utilisait \` pour délimiter les strings ? Ça permet d'utiliser `'` et `"` dans les strings qui reviennent plus souvent.
+Donc multi-line ? Ces deux bouts de code affichent la même chose :
+
+```
+print `one
+two
+three
+`
+```
+
+et
+
+```
+      print `one
+      two
+      three
+      `
+```
+
+sortie :
+
+```
+one
+two
+three
+```
+
+Ou alors on aligne sur le début du texte plutôt que le niveau d'identation (ça me paraît bcp plus lisible) ?
+
+```
+      print `one
+             two
+             three
+             `
+```
+
+## affectation / égalité
+
+Là on a plusieurs solutions :
+
+- deux symboles différents (habituellement `=` et `==`, parfois `:=`, `=` (Ada))
+- même symbole (`=`) mais la différence est liée au contexte, si on décide de ça, la syntaxe ne doit permettre aucune ambiguïté là-dessus
+- pas de symbole pour l'affectation (comme en asm), là encore en se basant sur le contexte, on peut qu'après un espace une valeur est attendue et sera affectée.
+
+  ```
+  a 1
+  b 2
+  if a + b = 3 then
+    print `obviously`
+  fi
+  ```
+
+  ```
+  robin [
+    name `Robin`
+    age 11
+    height 1.68
+    weight 51
+  ]
+  ```
+
+Je serais plus pour mettre deux symboles différents, ou en tout cas d'éviter à tout prix la réutilisation du symbole (qui n'a de fondement mathématique) pour éviter de perdre les débutants qui risquent d'encore moins discerner la distinction entre affecter et tester l'égalité.
+
+Proposition bold&dynamic : utiliser `is` pour affecter
+
+```
+address is `localhost`
+port is 8080
+c is a + b
+lost is lives < 0
+```
+
+On interdit les multi-affectations par ligne ? Type ce genre de choses :
+
+```
+a = 1, b = 2, c = 3 // JS style
+d, e, f = 4, 5, 6 // Go style
+
+i, j = j, i
+```
+
+Obligé d'écrire :
+
+```
+a = 1
+b = 2
+c = 3
+d = 4
+e = 5
+f = 6
+
+k = i
+i = j
+j = k
+```
+
+50 caractères vs 54 (8% de code en plus), osef ? En plus c'est rare d'inverser des valeurs donc le joli trick `i, j = j, i` on peut s'en passer non ?
+
+## opérateurs logiques
+
+Je préfère `and`, `or` et `not` à `&&`, `||` et `!` d'abord parce que ce sont des mots rarement utilisés pour déclarer quoi que ce soit, donc si c'est réservé c'est pas trop gênant, ensuite c'est également très court à écrire donc c'est pas plus difficile et ça permet d'utiliser un nom plutôt que des symboles dont il faut connaître le sens, de plus si on utilise des mots clefs les symboles deviennent éventuellement libres pour exprimer autre chose, car il va bien falloir en utiliser, à moins de vouloir écrire du Ada (déclaration d'un pointeur sur entier) :
+
```

Comment:
pareil je prefere `and`, `or`, `not`.

Apres faut reflechir a un moment a comment on appelle les fonctions, est-ce qu'on a des mot cle ou c'est des fonctions ? tout ca quoi.